File: ueueo-framework/ueueo-blob-storing/src/main/java/Abp/BlobStoring/BlobContainerConfigurations.cs
using System;
using System.Collections.Generic;
using JetBrains.Annotations;

namespace Volo.Abp.BlobStoring;

public class BlobContainerConfigurations
{
    private BlobContainerConfiguration Default => GetConfiguration<DefaultContainer>();

    private readonly Dictionary<String, BlobContainerConfiguration> _containers;

    public BlobContainerConfigurations()
    {
        _containers = new Dictionary<String, BlobContainerConfiguration>
        {
            //Add default container
            [BlobContainerNameAttribute.GetContainerName<DefaultContainer>()] = new BlobContainerConfiguration()
        };
    }

    public BlobContainerConfigurations Configure<TContainer>(
        Action<BlobContainerConfiguration> configureAction)
    {
        return Configure(
            BlobContainerNameAttribute.GetContainerName<TContainer>(),
            configureAction
        );
    }

    public BlobContainerConfigurations Configure(
        @NonNull String name,
        @NonNull Action<BlobContainerConfiguration> configureAction)
    {
        Check.NotNullOrWhiteSpace(name, nameof(name));
        Objects.requireNonNull(configureAction, nameof(configureAction));

        configureAction(
            _containers.GetOrAdd(
                name,
                () => new BlobContainerConfiguration(Default)
            )
        );

        return this;
    }

    public BlobContainerConfigurations ConfigureDefault(Action<BlobContainerConfiguration> configureAction)
    {
        configureAction(Default);
        return this;
    }

    public BlobContainerConfigurations ConfigureAll(Action<String, BlobContainerConfiguration> configureAction)
    {
        for (var container in _containers)
        {
            configureAction(container.Key, container.Value);
        }

        return this;
    }

    [NotNull]
    public BlobContainerConfiguration GetConfiguration<TContainer>()
    {
        return GetConfiguration(BlobContainerNameAttribute.GetContainerName<TContainer>());
    }

    [NotNull]
    public BlobContainerConfiguration GetConfiguration(@NonNull String name)
    {
        Check.NotNullOrWhiteSpace(name, nameof(name));

        return _containers.GetOrDefault(name) ??
               Default;
    }
}

File: ueueo-framework/ueueo-blob-storing/src/main/java/com/ueueo/blobstoring/BlobContainerConfigurations.java
package com.ueueo.blobstoring;

import org.springframework.lang.NonNull;
import org.springframework.util.Assert;

import java.util.HashMap;
import java.util.Map;
import java.util.Objects;
import java.util.function.BiConsumer;
import java.util.function.Consumer;

public class BlobContainerConfigurations {

    private BlobContainerConfiguration defaultConfiguration;

    private Map<String, BlobContainerConfiguration> containers;

    public BlobContainerConfigurations() {
        containers = new HashMap<>();
        containers.put(DefaultContainer.Name, new BlobContainerConfiguration(null));
        defaultConfiguration = getConfiguration(DefaultContainer.class);
    }

    public BlobContainerConfigurations configure(Class<?> containerType,
                                                 Consumer<BlobContainerConfiguration> configureAction) {
        BlobContainerNameAttribute attribute = containerType.getAnnotation(BlobContainerNameAttribute.class);
        return configure(attribute.name(), configureAction);
    }

    public BlobContainerConfigurations configure(
            @NonNull String name,
            @NonNull Consumer<BlobContainerConfiguration> configureAction) {
        Assert.hasLength(name, "name must not empty!");
        Objects.requireNonNull(configureAction);

        configureAction.accept(
                containers.computeIfAbsent(
                        name,
                        s -> new BlobContainerConfiguration(defaultConfiguration)
                )
        );

        return this;
    }

    public BlobContainerConfigurations configureDefault(Consumer<BlobContainerConfiguration> configureAction) {
        configureAction.accept(defaultConfiguration);
        return this;
    }

    public BlobContainerConfigurations configureAll(BiConsumer<String, BlobContainerConfiguration> configureAction) {
        for (Map.Entry<String, BlobContainerConfiguration> container : containers.entrySet()) {
            configureAction.accept(container.getKey(), container.getValue());
        }

        return this;
    }

    @NonNull
    public BlobContainerConfiguration getConfiguration(Class<?> containerType) {
        BlobContainerNameAttribute attribute = containerType.getAnnotation(BlobContainerNameAttribute.class);
        return getConfiguration(attribute.name());
    }

    @NonNull
    public BlobContainerConfiguration getConfiguration(@NonNull String name) {
        Assert.hasLength(name, "name must not empty!");

        return containers.get(name) != null ? containers.get(name) : defaultConfiguration;
    }
}
